src/main/java/com/iabtcf/utils/IntIterable.java

```diff
@@ -0,0 +1,32 @@
+package com.iabtcf.utils;
+
+/*-
+ * #%L
+ * IAB TCF Core Library
+ * %%
+ * Copyright (C) 2020 IAB Technology Laboratory, Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+/**
+ * An int primitive memory optimized iterable.
+ */
+public interface IntIterable extends Iterable<Integer> {
+    boolean contains(int value);
+
+    boolean containsAll(int... source);
+
+    IntIterator intIterator();
```

Comment:
I personally don't use iterators and have seen many people at my company favor streams lately. I'd love to have the toStream method on this interface purely to cut down on the amount of code to write to get that stream. I get it if we want to keep this as android friendly as possible, but we already include the IntStream in the utility class which might prevent the library from building in android as is.